feat: new payment flow explanation (#10)
portittech/pay-with-toonie-doc: README.md

- **REST API**: Authenticated Serverside integration to initiate the Payment Session to be passed to the client-side application
- **Vanilla JS SDK**: Client-side integration of the "Pay with Toonie Button" User Experience, able to handle QR generation, polling and communication with the payment API.

![Payment Buttons](imgs/buttons-page.png)

![Pay With Toonie](imgs/sample_qr.jpg)  ![Pay With Toonie](imgs/sample_qr_success.jpg) 

## REST API Integration

### Authentication

In order to initialize a new payment session, it is necessary to perform a very basic integration to an Authentication endpoint.

> Then you need to call the endpoint to create the payment session by passing some parameters, like an amount, a currency and a reason.

Reviewer: In the Authentication paragraph? I'd move this out

> You need to pass it a success and an error url too, appending to it the `?orderId={CHECKOUT_ORDER_ID}` which will be replaced in our system with the right value.

Reviewer: ```suggestion
You need to pass it a success and an error url too. You can use the `{PAYMENT_SESSION_ID}` placeholder anywhere in your url, which will be replaced in our system with the right value.
```

        "password": "customerpassword",
    })
});

// Create a payment session
const createPaymentSession = async (amount, currency, reason) => {
  const tokenData = await getTokenData();

  const res = await fetch("https://<ENVIRONMENT_API_URL>/acquiring/v1/payment", {
    method: "POST",
    headers: {

>       Authorization: ` Bearer ${tokenData.access_token}`,

Reviewer: ```suggestion
      "Authorization": `Bearer ${tokenData.access_token}`,
```

Reviewer 2: No need to add the `"`, it's js not json :)

      "content-type": "application/json",
    },
    body: JSON.stringify({

>       amount: amount ?? "1.23",

Reviewer: For this and all other values
```suggestion
      "amount": amount ?? "1.23",
```

      currency: currency ?? "EUR",
      reason: reason ?? "Test Payment 01",
      "successUrl": "https://www.success.com?orderId={CHECKOUT_ORDER_ID}",

>       "errorUrl": "https://www.error.com?orderId={CHECKOUT_ORDER_ID}",

Reviewer: ```suggestion
      "errorUrl": "https://www.yourdomain.com/error/?orderId={PAYMENT_SESSION_ID}",
```
